migrations/back_populate_uuids_plate_barcodes.py

```diff
@@ -0,0 +1,202 @@
+import logging
+import logging.config
+from datetime import datetime
+from typing import List
+
+from pymongo.collection import Collection
+
+from crawler.constants import (
+    COLLECTION_SAMPLES,
+    FIELD_COORDINATE,
+    FIELD_LH_SAMPLE_UUID,
+    FIELD_MONGODB_ID,
+    FIELD_PLATE_BARCODE,
+    FIELD_UPDATED_AT,
+    MLWH_LH_SAMPLE_UUID,
+    MLWH_MONGODB_ID,
+)
+from crawler.db.mongo import create_mongo_client, get_mongo_collection, get_mongo_db
+from crawler.sql_queries import SQL_MLWH_COUNT_MONGO_IDS, SQL_MLWH_GET_SAMPLE_FOR_MONGO_ID
+from crawler.types import Config, SampleDoc
+from migrations.helpers.shared_helper import (
+    extract_barcodes,
+    get_mongo_samples_for_source_plate,
+    mysql_generator,
+    validate_args,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+FIELD_UUID_UPDATED = "uuid_updated"
+
+
+class ExceptionSampleWithSampleUUID(Exception):
+    pass
+
+
+class ExceptionSampleCountsForMongoAndMLWHNotMatching(Exception):
+    pass
+
+
+"""
+Iterates over the list of plate barcodes provided in a CSV file.
+Finds all the samples in MongoDB with that plate barcode and iterates over those.
+Looks for the same sample in the lighthouse_sample table using mongodb_id and checks whether it has an lh_sample_uuid.
```

Comment:
About this, only an observation, that this is only valid for samples that were created while PSD had control on lighthouse_sample table, because the mongodb_id is not populated currently by GSU.

Reply:
I'll add that to the description.  It's useful to know for people looking at this in the future.